src/builder/commands/pip.rs

```diff
@@ -326,11 +341,18 @@ fn version_req(hash: &mut Digest, fname: &Path) -> Result<(), VersionError> {
                 if chunk.len() == 0 || chunk.starts_with("#") {
                     continue;
                 }
+                if let Some(req) = parse_req_filename(chunk) {
+                    included_files.push(req);
+                }
                 // Should we also ignore the order?
                 hash.item(chunk);
             }
             Ok(())
-    }).map_err(err)
+    }).map_err(err));
+    for req in &included_files {
```

Comment:
Any specific reason you don't do this recursively, but rather use a list? If there is, it's cute to insert it into the code as a comment.